### ISSUE TITLE
README.md - Can use the sample trained model

### ISSUE THREAD

**Issue by lovato** [RasaHQ/rasa_nlu#627]
**Rasa NLU version** (e.g. `0.7.3`):
0.10.0 w/ PIP
**Used backend / pipeline** (`mitie`, `spacy_sklearn`, ...):
None
**Operating system** (windows, osx, ...):
Linux
**Issue**:

Cannot work with a trained model, following readme instructions.

Whe using MD, is actually "train?name" or needs to be "train?project" like json format? They are different.

I used JSON. It worked at least for training. But I simply cannot make it to work from here. I keep getting:
`WARNING:rasa_nlu.project:Invalid model requested. Using default
`even with this status:
`{"available_projects": {"default": {"status": "ready", "available_models": ["fallback"]}, "test_model": {"status": "ready", "available_models": ["model_20171006-142730", "fallback"]}}}
`
Tks (ref #622)

**Content of configuration file** (if used & relevant):
```json

```

**Comment by PHLF** [comment]
The HTTP request structure is not related to the training format you use.
The warning is what it is: just a warning. If you didn't ask for a specific model when you sent a parse request targetting a project, Rasa will use the default model (the latest trained model) for this project.

**Comment by lovato** [comment]
Perhaps @tmbo can give also context here.
I am trying to go through README only and looks like there are a few updates needed there.

First, I trained the model. But I was unable to test it because this curl doesn't work:
```
$ curl 'http://localhost:5000/parse?q=hello&model=test_model'
{"error": "No project found with name 'default'."}
```
But, if I change to:
```
$ curl 'http://localhost:5000/parse?q=hello&project=test_model'
{"entities": [], "intent": {"confidence": 1.0, "name": "greet"}, "text": "hello"}
```
It works. Its the same problem fixed on #622 , the parameter "model" was replaced by "project".

Anyway, I still cannot go further, or I didn't get it. I am trying to do:
```
$ curl 'http://localhost:5000/parse?q=good+morning&project=test_model'
{"entities": [], "intent": {"confidence": 1.0, "name": "None"}, "text": "good morning"}
$ curl 'http://localhost:5000/parse?q=yeah&project=test_model'
{"entities": [], "intent": {"confidence": 1.0, "name": "None"}, "text": "yeah"}
```
Because I saw on the training json file that 'good morning' and 'yeah' is a valid input query.
https://raw.githubusercontent.com/RasaHQ/rasa_nlu/master/data/examples/rasa/demo-rasa.json
What am I missing here?

Just to share, this is /status
```
$ curl 'http://localhost:5000/status'
{"available_projects": {"test_model": {"status": "ready", "available_models": ["model_20171006-142730", "model_20171011-132722"]}}}
```
Its showing up twice since I did the training twice. Didn't know after a reset the model will remain.

Lovato

**Comment by PHLF** [comment]
You should have a look [at the docs](https://rasahq.github.io/rasa_nlu/http.html#serving-multiple-apps): you can ask a specific model for a specific project like this :

```bash
$ curl -XPOST localhost:5000/parse -d '{"q":"I am looking for Chinese food", "project":"my_restaurant_search_bot", "model":<model_XXXXXX>}'
```

> Didn't know after a reset the model will remain.

You can delete the corresponding folder under the "[path](https://rasahq.github.io/rasa_nlu/config.html#path)/test_model" directory. 

Moreover what did you use as a pipeline for training ? The `keyword` pipeline is very weak at classifying.

**Comment by lovato** [comment]
Hi. I always follow README steps before deep diving into documentation.

I tried your URL, making a few changes as I am seeing it on README.
```
curl -XPOST localhost:5000/parse -d '{"q":"yeah", "project":"test_model", "model":"model_20171006-142730"}'
{"entities": [], "intent": {"confidence": 1.0, "name": "None"}, "text": "yeah"}
```
Still not working anyways.
I am positive README still needs an update, as @tmbo did on #622. This is exactly what he asked for when he closed #622, to open a new ticket.

I even came up with this:
```
$ curl 'http://localhost:5000/parse?q=yeah&project=test_model&model=model_20171006-142730'
{"entities": [], "intent": {"confidence": 1.0, "name": "None"}, "text": "yeah"}
```

I understand that those requests were supposed to return a matching intent. Looks like even after training, only default built-in sample training can be used.

Does it make sense?

Best

**Comment by PHLF** [comment]
> I am positive README still needs an update, as @tmbo did on #622. This is exactly what he asked for when he closed #622, to open a new ticket.

Don't worry, I never said the opposite 😃 

> Still not working anyways.

What did you use as a pipeline for training your model ?

**Comment by lovato** [comment]
:-D

Well, pipelines are breifly mentioned, so ... I was not using it. I tried to add "pipeline=keyword" (just a  guess) at the end of my curl, just because there is one for /train. No success.

Taking (this time) a look into documentation, I was unable to find examples with pipelines, only a section describing how pipelines differ from each other. On "Using rasa NLU as a HTTP server" there is no mention to pipeline, and on "Processing Pipeline" there is no input examples, only outputs.

My test:
```
$ curl 'http://localhost:5000/parse?q=yeah&project=test_model&model=model_20171011-132722&pipeline=keyword'
{"entities": [], "intent": {"confidence": 1.0, "name": "None"}, "text": "yeah"}
```
Did the same for that "curl -XPOST" approach.
```
curl -XPOST localhost:5000/parse -d '{"q":"yeah", "project":"test_model", "model":"model_20171011-d132722", "pipeline":"keyword"}'
{"entities": [], "intent": {"confidence": 1.0, "name": "None"}, "text": "yeah"}
```
And I noticed that if I mess up with the project name, the curl results on an error. But if I do the same for the model, it keeps getting me back the same empty result, echoing the text I sent.

To give you context, I am working for past months with Alexa, API.ai, AWS, GH, etc. This is the first time I am going a step down into the AI engine, instead of just use something provided as a service. So I am not familiar with some concepts of ML yet (just a few, not all), which might be preventing me to have success. That's why a noob README helps in this case, because I can discover that my setup is right, and I can move on. I am almost sure I have a problem with my setup, but can't tell what.

I will also follow this one https://rasahq.github.io/rasa_nlu/tutorial.html in parallel.

**Comment by PHLF** [comment]
The pipeline is not really meant to be defined directly in the training request: Rasa NLU will use the one [defined](https://rasahq.github.io/rasa_nlu/pipeline.html#processing-pipeline) in your [configuration file](https://rasahq.github.io/rasa_nlu/config.html#pipeline). 
Having this set in a single global configuration file may be confusing and it would be better to have a configuration per project. I advise you to give a try to `spacy_sklearn` if you have spacy and scikit-learn installed.

**Comment by wrathagom** [comment]
Sorry I got lost a bit in the back and forth on what needs changed @lovato can you submit a PR with the proposed doc change?

**Comment by antoinecomp** [comment]
I don't know how much this may be linked but when I try to launch Rasa nlu as an http server with:

```
$ python -m rasa_nlu.server --path ~/Programing/Rasa/myflaskapp/models/nlu/default/moodnlu -e luis -c config_spacy.json
```

I got the `"No project found with name 'default'."` But it's where my model is with: `crf_model.pkl`, `intent_classifier_sklearn.pkl`, metadata.json`,  `training_data.json`

Otherwise if I try 

```
$ python -m rasa_nlu.server --path ~/Programing/Rasa/myflaskapp/models/nlu/ -e luis -c config_spacy.json
```

It definitely find the `/default` but even if I copy the content of `default/moodnlu` I got an `"Unable to initialize persistor"`. Do you have any idea?

**Comment by abhishakskilrock** [comment]
Hey everyone I am having the same issue, I run the command 
`curl -XPOST localhost:5050/parse -d '{"q":"hello there", "project":"/home/stpl/Bot/Weatherbot/models/nlu/default/weathernlu"}'`
Sometimes it works perfectly and showed me the results and entities and sometimes it showed me the following error:
 **{
  "error": "No project found with name '/home/stpl/Bot/Weatherbot/models/nlu/default/weathernlu'."
}**

Can anyone help me with this?

**Comment by mohamedHassanKa** [comment]
same error guys can any one help me
